pkg/bmh/baremetalhosts.go

```diff
@@ -0,0 +1,165 @@
+package bmh
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	bmhv1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+)
+
+const (
+	fiveScds time.Duration = 5 * time.Second
+)
+
+// BmhListBuilder provides struct for ClusterOperator List.
+type BmhListBuilder struct {
+	// Created baremetalhost object list
+	ObjectList []*BmhBuilder
+	// namespace baremetalhosts were created.
+	nsname string
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+	// Used in functions that define or mutate ClusterOperator definition. errorMsg is processed before the
+	// ClusterOperator object is created.
+	errorMsg string
+}
+
+// NewBmhListBuilder method creates new instance of Bmh List Builder.
+func NewBmhListBuilder(apiClient *clients.Settings, nsname string) *BmhListBuilder {
+	glog.V(100).Infof(
+		"Initializing new baremetalhosts list structure within %s namespace", nsname)
+
+	builder := &BmhListBuilder{
+		apiClient: apiClient,
+		nsname:    nsname,
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace for baremetalhosts is empty")
+
+		builder.errorMsg = "The namespace where baremetalhosts should be found cannot be empty"
+	}
+
+	return builder
+}
+
+// DiscoverList method gets the BaremetaHosts list in cluster and stores them in the builder struct.
+func (builder *BmhListBuilder) DiscoverList() error {
```

Comment:
Just call this Discover. Since it is a BmhListBuilder method, it's redundant to also use the word List in the method